### PR TITLE
Fix type-inference for toMap

### PR DESCRIPTION
**Pull request by sjakobi**
Previously the rule seemed to say that the type of a toMap expression is the type (kind) of its type annotation.

### CODE REVIEW

File: standard/type-inference.md [hunk 1]
```diff
@@ -766,7 +766,7 @@ of the record must have the same type.
     Γ ⊢ toMap e : List { mapKey : Text, mapValue : T₀ }
 
 
-    Γ ⊢ e :⇥ {}    Γ ⊢ T₀ :⇥ List { mapKey : Text, mapValue : T₁ }
+    Γ ⊢ e :⇥ {}   T₀ ⇥ List { mapKey : Text, mapValue : T₁ }
```
Comment (sjakobi) [inline standard/type-inference.md]: Not sure this is quite right yet…

Something like this maybe?
```suggestion
    Γ ⊢ e :⇥ {}   Γ ⊢ T₀ ⇥ List { mapKey : Text, mapValue : T₁ }
```

Comment (Gabriel439) [inline standard/type-inference.md]: I think this is what you want:

```suggestion
    Γ ⊢ e :⇥ {}   Γ ⊢ T₀ :⇥ Type   T₀ ⇥ List { mapKey : Text, mapValue : T₁ }
```

Comment (sjakobi) [reply standard/type-inference.md]: Thanks! :)

Comment (Gabriel439) [reply standard/type-inference.md]: You're welcome! :slightly_smiling_face:

**Comment by Gabriel439** [comment]
@f-f: Could you review this when you have a chance?  It's not strictly necessary, but I'd like to get in this fix if possible before cutting the 9.0.0 release tomorrow

**Pull request merged**